pkg/controllers/management/nodepool/util.go

```diff
@@ -0,0 +1,52 @@
+package nodepool
+
+import (
+	"strconv"
+
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+
+	"github.com/sirupsen/logrus"
+)
+
+// use pointer to avoid copying large number of structs
+
+type byNodeRequestedHostName []*v3.Node
+
+/*
+ Expecting to see strings styled [Prefix] + [some decimal value] which means we can do some specific things
+Inspired by https://github.com/facette/natsort/blob/2cd4dd1e2dcba4d85d6d3ead4adf4cfd2b70caf2/natsort.go
+*/
+
+func (s byNodeRequestedHostName) Len() int {
+	return len(s) //returns len of slice of structs
+}
+
+func (s byNodeRequestedHostName) Less(a, b int) bool {
+	return Compare(s[a].Spec.RequestedHostname, s[b].Spec.RequestedHostname)
+}
+
+func (s byNodeRequestedHostName) Swap(a, b int) {
+	s[a], s[b] = s[b], s[a]
+}
+
+// Compare returns true if the first string precedes the second one according to natural order
+func Compare(a, b string) bool {
+	s1 := nameRegexp.FindStringSubmatch(a)
```

Comment:
FWIW in the UI we compute (a string](https://github.com/rancher/ui/blob/3e647571b00187f38e791ea9ef15dfc75f4ba981/lib/shared/addon/utils/util.js#L213-L223) that is ascii-sortable once by zero-padding number suffixes (with regex) to a fixed long-enough length (i.e. `thing1` -> `thing0000000001`) and then let the normal native implementation sort that.

I have no idea how slow regex, atoi, etc are in Go but something like 2*n*log(n) extra operations the same strings many times is significant to the UI (although I do sorts much more often, and only have to compute the string once per instance of a resource for the life of the page)

Reply:
https://github.com/rancher/rancher/blob/6a0e89c1012c33c520729929143b1c032e22c533/pkg/controllers/management/nodepool/nodepool.go#L258
It looks like the backend should be doing this as well... Since we follow that convention for naming